### PR TITLE
Read MCS file from SAPHIRE

### LINKED ISSUE #39: update MCS reader to handle SAPHIRE output
--------
Issue Description
--------

**Is your feature request related to a problem? Please describe.**
SAPHIRE will output MCS in its own format, since SR2ML is using MCS to create margin solver, it will be useful to directly process SAPHIRE output to extract the MCS for margin solver. 

**Describe the solution you'd like**
update the current MCS reader.

**Describe alternatives you've considered**

**Additional context**

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by wangcj05**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
Update the current MCS reader to process SAPHIRE MCS output file and close #39 

##### What are the significant changes in functionality due to this change request?

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual.
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Regression tests have to complete successfully.
- [x] 5. If significant functionality is added, there must be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test.
- [x] 6. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.

### CODE REVIEW

File: src/PostProcessors/MCSimporter.py [hunk 1]
```diff
@@ -147,20 +148,30 @@ def mcsReader(mcsListFile):
...
+        elementsList = l.split(',')
+        # skip empty line
+        if elementsList[0].strip() == '':
+          continue
+        mcsIDs = np.append(mcsIDs,elementsList[0])
+        probability=np.append(probability, float(elementsList[1]))
+        # skip column 3 which is the fraction to the total
+        mcs = list(element.strip() for element in elementsList[3:])
+        mcsList.append(mcs)
+        beList.update(mcs)
```
Comment (mandd) [inline src/PostProcessors/MCSimporter.py:179]: is this line correct? mcsList and beList are the same at the end

**Comment by mandd** [comment]
Does the manual need to be updated after these changes?

File: src/PostProcessors/MCSimporter.py (hunk 1 above)
Comment (wangcj05) [reply src/PostProcessors/MCSimporter.py:179]: I think there are different. Here, mcsList is a list of list minimal cut sets, while beList is just a set of basic events. beList was previous defined as beList = set(). The name is confusing, but it was used in the original code.

**Comment by wangcj05** [comment]
> Does the manual need to be updated after these changes?
We are currently just doing the unit testing here. I think it is not necessary to update the manual now. If you are fine with this PR, I will update MCSSolver and MCSImporter to have the optional input from the user to select the type of MCS file. In that case, we need to add additional tests and update the manual.

**Comment by wangcj05** [comment]
@mandd This PR is ready for your second review.

**Review by mandd — APPROVED** [review]

**Comment by mandd** [comment]
Test board is green. Merging.

**Pull request merged**